### PR TITLE
Only enable TransactionScopeSuppressFeature if the transport selection if of the current endpoint matches the selected TransportType

### PR DESCRIPTION
**Pull request by danielmarbach**
Suppress behavior could interfere with other transports in multi-endpoint hosting scenarios

### CODE REVIEW

File: src/Transport/Configuration/SettingsKeys.cs [hunk 1]
```diff
@@ -13,5 +13,6 @@ static class SettingsKeys
         public const string RuleNameShortener = Base + nameof(RuleNameShortener);
         public const string TransportType = Base + nameof(TransportType);
         public const string CustomTokenProvider = Base + nameof(CustomTokenProvider);
+        public const string Self = Base + "1BA47B92-6987-4B9C-AFF3-B906B9FED0B3";
```
Comment (SeanFeldman) [inline src/Transport/Configuration/SettingsKeys.cs]: `Self` - could you please provide a more meaningful name? Without reading the [comment](https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/blob/f33a5b9929a6bd663c8eeaac7359fb6529956607/src/Transport/Receiving/TransactionScopeSuppressFeature.cs) it's hard to determine what this key is for.

Comment (danielmarbach) [reply src/Transport/Configuration/SettingsKeys.cs]: Feel free to change it to your needs. We couldn't come up with a better name. Sorry

**Comment by bording** [comment]
There's an alternative approach to this that we used in the RabbitMQ transport that I think would be a better way to handle this: https://github.com/Particular/NServiceBus.RabbitMQ/pull/472

**Comment by danielmarbach** [comment]
Like that. Simple. Forget that these SettingsExtensions exist

**Comment by danielmarbach** [comment]
@bording @SeanFeldman changed the approach. Have a look

**Comment by SeanFeldman** [comment]
Replaced by #12

**Pull request closed without merging**